src/term/term_to_net.rs

```diff
@@ -316,22 +327,45 @@ impl Op {
 pub struct LabelGenerator {
   untagged_dups: u32,
   tagged_dups: HashMap<Name, u32>,
+  untagged_sups: u32,
+  tagged_sups: HashMap<Name, u32>,
+}
+
+enum LabelFor {
+  Dup,
+  Sup,
 }
 
 impl LabelGenerator {
   // If some tag and new generate a new label, otherwise return the generated label.
   // If none use the implicit label counter.
-  fn generate(&mut self, tag: &Option<Name>) -> u32 {
+  fn generate(&mut self, tag: &Option<Name>, label_for: LabelFor) -> u32 {
     if let Some(tag) = tag {
-      let next_lab = self.tagged_dups.len() as u32;
-      match self.tagged_dups.entry(tag.clone()) {
+      let tagged = self.get_tagged(label_for);
+      let next_lab = tagged.len() as u32;
+      match tagged.entry(tag.clone()) {
         Entry::Occupied(e) => *e.get(),
         Entry::Vacant(e) => *e.insert(next_lab),
       }
     } else {
-      let lab = self.untagged_dups;
-      self.untagged_dups += 1;
+      let untagged = self.get_untagged(label_for);
+      let lab = *untagged;
+      *untagged += 1;
```

Comment:
This is what I meant. If dups and sups can share the same labels depending on the order they appear in the program, we get unpredictable runtime results.